Assets/Scripts/Framework/puzzels/SequencePuzzle/ColorSequenceManager.cs

```diff
@@ -0,0 +1,117 @@
+using UnityEngine;
+using System;
+using UnityEngine.Events;
+using System.Collections.Generic;
+
+public sealed class ColorSequenceManager : MonoBehaviour, SequenceManager<EnumColor>
+{
+    public EnumColor[] Sequence { get; set; }  = new EnumColor[5];
+    public EnumColor[] SubmittedSequence { get; set; } = new EnumColor[5];
+
+    [SerializeField] private UnityEvent onSolved = new UnityEvent();
+    [SerializeField] private UnityEvent onReset = new UnityEvent();
+
+
+    
+
+    private int sequenceLength = 3;
+
+    #region Manage Sequence
+    public void StartSequence()
+    {
+        for (int i = 0; i < sequenceLength; i++)
+        {
+            EnumColor color;
+            color = GetRandomColor();
+            while(IsColorInSequence(color) || color == EnumColor.Unassigned)
+            {
+                color = GetRandomColor();
+            }
+            Sequence[i] = color;
+        }
+    }
+    public void NextRound()
+    {
+        sequenceLength++;
+        for (int i = 0; i < sequenceLength; i++)
+        {
+            if (Sequence[i] != EnumColor.Unassigned) continue;
+
+            EnumColor color;
+            color = GetRandomColor();
```

Comment:
```suggestion
           EnumColor color = GetRandomColor();
```
Creating en Assignment samen, dit is meer preference maar ik denk tenzij je vooraf iets moet checken zoals null check of string.empty checken bij string dan kan je meestal gwn samen doen